marlowe-test/src/Spec/Marlowe/Semantics/Arbitrary.hs

```diff
@@ -291,13 +298,10 @@ randomPubKeyHashes =
   , "a2bd7dd7f41c2781d1d11c7f4994fac750525705f9c259f97cb27d0e"
   , "c5b4c543a0d0d181ec387ad8250b18617bb18bcf2eccc0f27fe7aa23"
   , "d877b83ece77d785fee4a52bd7226949fa64e111aa0e20cd4a1c471b"
-  , "e14025a93f867851b9bb3c48601d1845bcbe9e2e1856c16cfc0522"      -- NB: Too short for ledger.
-  , "e3351d289f3eaa66e500f17b91a74e492193f4485c32e5ad606da83542"  -- NB: Too long for ledger.
   ]
 
 instance Arbitrary PubKeyHash where
-  arbitrary = arbitraryFibonacci randomPubKeyHashes `suchThat` (\(PubKeyHash h) -> lengthOfByteString h == 28)
-  shrink x = filter (< x) randomPubKeyHashes
+  arbitrary = elements randomPubKeyHashes
```

Comment:
- simplified
- avoid use of `suchThat`

Reply:
This removes shrinking, I think it was useful to have it as it was, because, then you get counterexamples with fewer different addresses (if the actual address doesn't matter), easier to read

Reply:
That may be, but it assumes the context of the test (which arbitrary instances shouldn't do). Shrinking should produce structurally smaller values that can be derived from the value being shrunk, and picking a different element from the "universe" of hashes A) isn't structurally smaller and B) isn't creating a "smaller version" of the value being shrunk - it's just returning a new, unrelated value.